src/FindDependencies.c

```diff
@@ -249,6 +249,31 @@ static bool RuleBestThanLatest(char *candidate, char *latest)
 }
 
 
+static FILE *GetManagerRulesFromAlien(struct parse_data *data, struct search_options *options)
+{
+  char *sp;
+  char alien[LINE_MAX];
+  char aliencmd[LINE_MAX+32];
+  FILE *fp;
+
+  sp = strchr(data->depname, ':');
+  if (!sp) {
+    WARN(options, "WARNING: %s is not an Alien dependency, ignoring dependency.\n", data->depname);
+    return NULL;
+  }
+  strncpy(alien, data->depname, sp-data->depname);
+  alien[sp - data->depname] = 0;
+
+  snprintf(aliencmd, sizeof(aliencmd)-1, "Alien-'%s' --get-manager-rule", alien);
+  fp = popen(aliencmd, "r");
```

Comment:
It seems possible that an ill-formed dependency in data could result in code execution here, although I'm not sure if that's a real problem worth worrying about. It requires a dependency entry like `LuaRocks';rm<TAB>rf<TAB>/;'` (since spaces and redirection characters are already ruled out by earlier processing). This is probably also possible within the package name itself already, and review should avoid it ever coming up. I don't think anything needs to be done about it, but I want to raise it in case someone has more concerns.

Reply:
We have a couple more instances where `popen()` is passed a buffer whose contents come from the Dependencies file. I think it's fine to merge this as-is and then replace those calls on a follow-up PR with a function that wraps `fork()` + `exec()` + `wait()`.